ouroboros-consensus-test/bench-mempool/Main.hs

```diff
@@ -0,0 +1,65 @@
+{-# LANGUAGE NumericUnderscores #-}
+{-# LANGUAGE OverloadedLists    #-}
+
+
+module Main (main) where
+
+import           Control.Monad (void)
+import           Data.Set ()
+import           Test.Tasty.HUnit (testCase, (@?=))
+
+import qualified Control.Tracer as Tracer
+import           Test.Tasty.Bench (bench, bgroup, defaultMain, nfIO)
+
+import           Bench.Consensus.Mempool
+import           Bench.Consensus.Mempool.TestBlock (TestBlock)
+import qualified Bench.Consensus.Mempool.TestBlock as TestBlock
+import           Bench.Consensus.MempoolWithMockedLedgerItf
+import qualified Ouroboros.Consensus.Mempool.Capacity as Mempool
+
+main :: IO ()
+main = defaultMain [ benchmarkJustAddingTransactions ]
+  where
+    benchmarkJustAddingTransactions =
+        bgroup "Just adding transactions"
+          $  fmap benchAddNTxs [1_000, 10_000, 100_000, 1_000_000]
+          <>
+             [ testCase "Transactions are added" $ do
+                 let
+                   n   = 1000
+                   txs = mkNTryAddTxs 1000
+                 mempool    <- addNTxs txs
+                 mempoolTxs <- getTxs mempool
+                 mempoolTxs @?= txsAddedInCmds txs
+                 length mempoolTxs @?= n
+             ]
+      where
+        benchAddNTxs n = bench (show n) $ nfIO $ addNTxs' $ mkNTryAddTxs n
```

Comment:
If the cost of adding the txs is what you want to benchmark, and not opening the mocked mempool or creating the mocked transactions you want to add, I would suggest using [`env`/`envWithCleanup`](https://hackage.haskell.org/package/tasty-bench-0.3.3/docs/Test-Tasty-Bench.html#v:env) to do that work before running the benchmark, and then only calling the function-under-bench with those inputs. The inputs will by then already have been forced to normal form by `env`, such that the benchmark only measures the time spent adding the txs. Using `env` might not be strictly useful if the setup is much cheaper than adding the txs, but I don't think it hurts to use `env` regardless

It would look something like:
```suggestion
        benchAddNTxs n = env (doSetup n) $ \(mempool, txs) ->
           bench (show n) $ nfIO $ addNTxs' mempool 
```

Reply:
Absolutely! I overlooked this. Thank you :pray:

Reply:
Hmm, `env`'s documentation recommends using [`withResource`](https://hackage.haskell.org/package/tasty-1.4.3/docs/Test-Tasty.html#v:withResource), which does not seem to perform any normal form evaluation.

Reply:
I hope I never forget to use a function like `env` when benchmarking. 

Before `env`:

![image](https://user-images.githubusercontent.com/175315/221574669-0af9e73b-fc57-45c6-904e-c63c50bd07cc.png)

With `env`:

![image](https://user-images.githubusercontent.com/175315/221574730-3a024154-b76b-4f41-860c-5bad153c2682.png)

Reply:
Generating transactions looked relatively cheap, so maybe opening the mempool was the costly part